mm2src/common/log.rs

```diff
@@ -183,6 +184,78 @@ macro_rules! log {
     }}
 }
 
+/// Log to the `ctx` dashboard with single tags, or key-value tags, or without any tags.
+///
+/// # Examples
+///
+/// ## With single and key-value tags
+///
+/// ```rust
+/// log_tag!(
+///   ctx,
+///   "😅",
+///   "tx_history",
+///   "coin" => coin.as_ref().conf.ticker;
+///   fmt = "Some message: {}",
+///   any_message
+/// );
+/// ```
+///
+/// ## Without any tags
+///
+/// ```rust
+/// log_tag!(ctx, "😅"; fmt = "Some message: {}", any_message);
+/// ```
+///
+/// # Important
+///
+/// Don't forget to separate tags and message formatting using `;` symbol.
+#[macro_export]
+macro_rules! log_tag {
+    ($ctx:expr, $emotion:literal $(, $tag_key:expr $(=> $tag_val:expr)? )* ; fmt = $($arg:tt)*) => {{
+        let tags: &[&dyn $crate::log::TagParam] = &[
+            $(
+                &(
+                    $tag_key.to_string()
+                    $(, $tag_val.to_string())?
+                )
+            ),*
+        ];
+        let line = ERRL!($($arg)*);
+        $ctx.log.log($emotion, tags, &line);
+    }};
+}
+
+#[macro_export]
+macro_rules! error_if {
```

Comment:
Yes, I thought exactly as you, but what stopped me to integrate it:
`crate::log::warn!("{:?}", e)` doesn't use the `track_caller` feature, so its location will be always a line within `common.rs`.
We can do something like this:
```rust
if let Err(e) = self {
    let caller = Location::caller();
    crate::log::warn!("{}:{}] {:?}", caller.filename(), caller.line(), e);
}
```
But it will generate log lines like this:
`common:123] real_file:456] An error`

Reply:
> We can do something like this

I think it is an acceptable solution :slightly_smiling_face:

Reply:
Done